walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,100 @@
 
 package com.google.sps;
 
+import com.google.sps.Event;
+import com.google.sps.TimeRange;
+import com.google.sps.MeetingRequest;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  public Collection<TimeRange> query(Collection<Event> events, 
+                                      MeetingRequest request) {
+    // Get unavailable time for all required attendees in meeting request
+    ArrayList<TimeRange> unavailable = new ArrayList<>();
+    for (String attendee: request.getAttendees()) {
+      ArrayList<TimeRange> meetingTimes = getMeetingTimes(events, attendee);
+      unavailable = merge(unavailable, meetingTimes);
+    }
+
+    Collection<TimeRange> result = new ArrayList<>();
+    for (TimeRange t: getAvailability(unavailable)) {
+      // Get time ranges that are longer than the requested meeting duration
+      if (t.duration() >= request.getDuration()) {
+        result.add(t);
+      }
+    }
+
+    return result;
+  }
+
+  private ArrayList<TimeRange> merge(ArrayList<TimeRange> result, 
+                                      ArrayList<TimeRange> other) {
```

Comment:
the behaviour of this function is unclear. please add documents.
note that java always pass Object by reference, which means you would modify result in place.
In some cases, the result is modified and set to the returned result. but in some other cases, it's left unchanged.

Reply:
Since object is passed by reference, is it okay to make this function return void and modify result in place?
Or is it a better to always create a new local variable and return that local variable?

Reply:
I think both are okay, but you should make it clear.